core-bundle/src/DataContainer/DataContainerOperation.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\DataContainer;
+use Contao\StringUtil;
+
+class DataContainerOperation implements \ArrayAccess
+{
+    private string|null $html = null;
+
+    /**
+     * @internal
+     */
+    public function __construct(private readonly string $name, private array $operation, private readonly array $record, private readonly DataContainer $dataContainer)
+    {
+        $id = StringUtil::specialchars(rawurldecode((string) $record['id']));
+
+        if (isset($operation['label'])) {
+            if (\is_array($operation['label'])) {
+                $operation['title'] = sprintf($operation['label'][1] ?? '', $id);
+                $operation['label'] = $operation['label'][0] ?? $name;
+            } else {
+                $operation['label'] = $operation['title'] = sprintf($operation['label'], $id);
+            }
+        } else {
+            $operation['label'] = $operation['title'] = $name;
+        }
+
+        $attributes = !empty($operation['attributes']) ? ' '.ltrim(sprintf($operation['attributes'], $id, $id)) : '';
+
+        // Add the key as CSS class
+        if (str_contains($attributes, 'class="')) {
+            $attributes = str_replace('class="', 'class="'.$name.' ', $attributes);
+        } else {
+            $attributes = ' class="'.$name.'" '.$attributes;
+        }
+        $operation['attributes'] = $attributes;
+
+        $this->operation = $operation;
```

Comment:
```suggestion
```
This should already be assigned to the implicitly declared property.

Reply:
changed in b5d188873ed4e1146dcfad08f2a98154235c3ae3